test/NameRegistry.upgrade.t.sol

```diff
@@ -68,6 +68,31 @@ contract NameRegistryUpgradeTest is Test {
         assertEq(nameRegistryV2Proxy.number(), 42);
     }
 
+    function testUpgradeMaintainsExistingRoles(
+        bytes32 role1,
+        bytes32 role2,
+        bytes32 role3,
+        address alice,
+        address bob,
+        address charlie
+    ) public {
+        _grant(role1, alice);
+        _grant(role2, bob);
+
+        vm.assume(!nameRegistryProxy.hasRole(role3, charlie));
```

Comment:
nit: i think you can simplify it to just a single role, and alice having the role and bob not having one, unless there is some codepath that won't be tested if we simplify

Reply:
I think the same codepaths would be tested if we omitted a second granted role, but I think testing the successful migration of multiple storage slots is a bit more implementation agnostic, although I agree paranoid 😂